src/block/Button.php

```diff
@@ -91,4 +89,19 @@ public function onNearbyBlockChange() : void{
 	private function canBeSupportedAt(Block $block, int $face) : bool{
 		return $block->getAdjacentSupportType(Facing::opposite($face))->hasCenterSupport();
 	}
+
+	public function onProjectileInteraction(Projectile $projectile) : void{
+		if($projectile instanceof WindCharge) {
```

Comment:
need to check if its not pressed

Reply:
$this->pressed(); in this context, thereby extending the pushed duration if already down, I do not see a need to change this, thoughts?

Reply:
by checking that it is not activated, you will avoid extra sound.